### PR TITLE
Adding README for Sidekiq instrumentation

### PR DESCRIPTION
**Pull request by AzfaarQureshi**
This PR adds documentation for OpenTelemetry's Sidekiq Instrumentation, closing #241 . The README describes the Sidekiq gem, installation methods and its license, following the pattern set in [instrumentation/sinatra](https://github.com/open-telemetry/opentelemetry-ruby/blob/master/instrumentation/sinatra/README.md) and [instrumentation/concurrent_ruby](https://github.com/open-telemetry/opentelemetry-ruby/blob/master/instrumentation/concurrent_ruby/README.md) for consistency.

### CODE REVIEW

**Review by mwear — COMMENTED** [review]
👋 Welcome and thanks for the contribution. This looks good, but I'm wondering if we should point people towards the example (in the examples) folder. WDYT?

**Comment by AzfaarQureshi** [comment]
@mwear Good idea! I added a small Examples section. Is this along the lines of what you were thinking?

File: instrumentation/sidekiq/README.md
Comment (AzfaarQureshi) [inline instrumentation/sidekiq/README.md:34]: @mwear is it okay to add a link to the actual file here or is that considered bad practice considering it might become dead if there is ever a code restructure?

Comment (ericmustin) [reply instrumentation/sidekiq/README.md:34]: I think it's ok fwiw, will defer to @mwear on any formatting prefs. nice work here @AzfaarQureshi

File: instrumentation/sidekiq/README.md
Comment (ericmustin) [inline instrumentation/sidekiq/README.md:48]: can remove, dupe
```suggestion

```

**Review by ericmustin — CHANGES_REQUESTED** [review]
very minor formatting changes but otherwise lgtm, nice work @AzfaarQureshi

File: instrumentation/sidekiq/README.md
Comment (fbogsany) [reply instrumentation/sidekiq/README.md:34]: 👍 this is fine - if we restructure the code, we'll also fix the documentation.

**Review by fbogsany — APPROVED** [review]
LGTM after @ericmustin's change. Thanks @AzfaarQureshi!

**Review by robertlaurin — APPROVED** [review]

**Comment by AzfaarQureshi** [comment]
@ericmustin oops cant believe I missed the dupe. Thanks for the catch! Its fixed and ready for re-review 😄

**Review by ericmustin — APPROVED** [review]
:shipit: 👍

**Comment by fbogsany** [comment]
Can you please rebase @AzfaarQureshi ?

**Review by mwear — APPROVED** [review]
Thanks @AzfaarQureshi!

File: instrumentation/sidekiq/README.md
Comment (mwlang) [inline instrumentation/sidekiq/README.md:48]: bundler-home listed twice.  Everything looks great.

Comment (fbogsany) [reply instrumentation/sidekiq/README.md:48]: Thanks for catching that - it was a problem in another PR too, but I was lulled into ignorance after reviewing a few that *didn't* have that problem.

**Pull request merged**
